app/components/primer/alpha/select_panel_element.ts

```diff
@@ -0,0 +1,1030 @@
+          return
+        }
+
+        break
+      }
+    }
+  }
+
+  #toggleIncludeFragmentElements(showError: boolean) {
+    for (const el of this.includeFragment.querySelectorAll('[data-show-on-error]')) {
+      if (el instanceof HTMLElement) el.hidden = !showError
+    }
+    for (const el of this.includeFragment.querySelectorAll('[data-hide-on-error]')) {
+      if (el instanceof HTMLElement) el.hidden = showError
+    }
+  }
+
+  #handleRemoteInputEvent(event: Event) {
+    switch (event.type) {
+      case 'remote-input-success': {
+        this.#clearErrorState()
+        this.#updateItemVisibility()
+        this.#checkSelectedItems()
+        break
+      }
+
+      case 'remote-input-error': {
+        this.bodySpinner?.setAttribute('hidden', '')
+
+        if (this.includeFragment || this.visibleItems.length === 0) {
+          this.#setErrorState(ErrorStateType.BODY)
+        } else {
+          this.#setErrorState(ErrorStateType.BANNER)
+        }
+
+        break
+      }
+
+      case 'loadstart': {
+        if (!this.#performFilteringLocally()) {
+          this.#clearErrorState()
+          this.bodySpinner?.removeAttribute('hidden')
+
+          if (this.bodySpinner) break
+          this.#setTextFieldLoadingSpinnerTimer()
+        }
+
+        break
+      }
+
+      case 'loadend': {
+        this.#filterInputTextFieldElement.hideLeadingSpinner()
+        if (this.#loadingAnnouncementTimeoutId) clearTimeout(this.#loadingAnnouncementTimeoutId)
+        if (this.#loadingDelayTimeoutId) clearTimeout(this.#loadingDelayTimeoutId)
+        this.dispatchEvent(new CustomEvent('loadend'))
+        break
+      }
+    }
+  }
+
+  #defaultFilterFn(item: HTMLElement, query: string) {
+    const text = (item.getAttribute('data-filter-string') || item.textContent || '').toLowerCase()
+    return text.indexOf(query.toLowerCase()) > -1
+  }
+
+  #handleSearchFieldEvent(event: Event) {
+    if (event.type === 'keydown' && (event as KeyboardEvent).key === 'ArrowDown') {
+      if (this.focusableItem) {
+        this.focusableItem.focus()
+        event.preventDefault()
+      }
+    }
+    if (event.type !== 'input') return
+
+    // remote-input-element does not trigger another loadstart event if a request is
+    // already in-flight, so we use the input event on the text field to reset the
+    // loading spinner timer instead
+    if (!this.bodySpinner && !this.#performFilteringLocally()) {
+      this.#setTextFieldLoadingSpinnerTimer()
+    }
+
+    if (this.#fetchStrategy === FetchStrategy.LOCAL || this.#fetchStrategy === FetchStrategy.EVENTUALLY_LOCAL) {
+      if (this.includeFragment) {
+        this.includeFragment.refetch()
+        return
+      }
+
+      this.#updateItemVisibility()
+    }
+  }
+
+  #updateItemVisibility() {
+    if (!this.list) return
+
+    let atLeastOneResult = false
+
+    if (this.#performFilteringLocally()) {
+      const query = this.filterInputTextField?.value ?? ''
```

Comment:
✅ Thanks for capturing this late TS warning [fix](https://github.com/github/github/pull/332814/files)!